aws_lambda_powertools/utilities/batch/sqs.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+
+"""
+Batch SQS utilities
+"""
+
+from typing import List, Optional
+
+import boto3
+from botocore.config import Config
+
+from aws_lambda_powertools.middleware_factory import lambda_handler_decorator
+
+from .base import BasePartialProcessor
+
+
+class PartialSQSProcessor(BasePartialProcessor):
+    def __init__(self, config: Optional[Config] = None):
+        config = config or Config()
+        self.client = boto3.client("sqs", config=config)
+        self.success_messages: List = []
+        self.fail_messages: List = []
+
+        super().__init__()
+
+    def get_queue_url(self):
+        """
+        Format QueueUrl from first records entry
+        """
+        if not getattr(self, "records", None):
+            return
+
+        *_, account_id, queue_name = self.records[0]["eventSourceARN"].split(":")
+        return f"{self.client._endpoint.host}/{account_id}/{queue_name}"
+
+    def get_entries_to_clean(self):
```

Comment:
```suggestion
    def _get_entries_to_clean(self) -> List:
```

Other than a unit test there is no need to expose this method right?

Reply:
Absolutely. They were private methods in my original approach, changed for some reason but forgot to go back, thx.

About the annotation, I'm about to review everything to write docs.